### ISSUE TITLE
Sync between devices

### ISSUE THREAD

**Issue by arn0nym0us** [futo-org/grayjay-android#1306]
### Use Cases

When using Grayjay on multiple devices there is no way to sync subscriptions, playlists and watch history between the devices. Except for manually doing imports and exports of the data.

### Proposal

Would be great if there was a sync feature kind of like with Brave browser where u can link up devices using a Passphrase and then they synchronise their encrypted data via a cloud server.
An other solution could be to provide the ability to connect and sync with Invidious and Piped Servers, but that would only work for YouTube content since other providers aren't supported there.

**Comment by Kelvin-FUTO** [comment]
We are already working on a system that does something like this between mobile & desktop, but the protocol is not really device biased, so they should be able to support mobile-mobile as well once that is done.

**Comment by jroddev** [comment]
This is the big blocker for me to use GrayJay at the moment.
At home I use either my Android Tablet or Android TV, but when I'm out and about I use my Android Phone.
For my case I would be happy to just sync the data via NextCloud/WebDAV([example](https://blog.jroddev.com/webdav-for-application-sync/)), or an S3-like object storage.

@Kelvin-FUTO are there any more details on what you're planning the Syncing System?

**Comment by cyraid** [comment]
@jroddev yup exactly. Same with me. I use Smart Tube which was way easier to setup YouTube and didn't have to scarily enter my password in the app (would be nice if the YouTube plugin just allowed you to login with the device or a different browser).

Back on track though: I find myself watching on smart tube on the TV, then going to grayjay, and then YT on my desktop (with ads disabled).

Would be nice if YT plugin would pull the history on start or even periodically.

**Comment by support-grayjay** [comment]
This will be included when Desktop application is released. Soooon.

**Comment by Zvonimir-FUTO** [comment]
This has been added in the latest version of Grayjay.

**Comment by cyraid** [comment]
This doesn't really solve when I watch a video on my TV. It doesn't show up in gray jay. You need that import of previous history from sources themselves.

**Comment by Zvonimir-FUTO** [comment]
It would be nice if you would open a new issue for this

**Comment by cyraid** [comment]
> It would be nice if you would open a new issue for this

[Done.](https://github.com/futo-org/grayjay-android/issues/1569)

**Comment by jroddev** [comment]
@Zvonimir-FUTO is there any documentation on this feature? I'm not sure how to use it.
Is it the Sync Settings option? I set that up but subscriptions + playlists are still different.

**Comment by cyraid** [comment]
I think you have to have them both running at the same time? Some status of that would be nice. Like, synced "just now" and "syncing". A sync now button would be nice too.

**Comment by Zvonimir-FUTO** [comment]
@jroddev apologies for late reply, no idea how I missed this. Sync runs in background and is triggered as soon as any change is made and sync is super fast since it sync data is very small so there is no point of progress bar and you would be spammed with "just now" sync messages for every change.

Something that makes this confusing for users is that view doesn't refresh immediately which we plan on working so you might not see the changes until you refresh the view or reopen the app.

Another thing users often confuse is Grayjay sync (local data) and syncing with plugin playlists like Youtube. Latter is not supported, yet.

**Comment by Zvonimir-FUTO** [comment]
I think there are small pop up notifications on desktop when something is sync if I remember correctly. As for the Youtube issue, appears to be a bug. Please open a new issue for it here if it wasn't opened already: https://github.com/futo-org/Grayjay.Desktop/issues

**Comment by ykuksenko** [comment]
@Zvonimir-FUTO I was able to successfully sync playlists between phone and desktop. However when items are deleted from a playlist this is not reflected on the other device. Is this the plugin playlist function you said does not work? Or does this need a new issue opened? (I have tried restarting the apps and mobile device.)

Thanks.

**Comment by Zvonimir-FUTO** [comment]
> @Zvonimir-FUTO I was able to successfully sync playlists between phone and desktop. However when items are deleted from a playlist this is not reflected on the other device. Is this the plugin playlist function you said does not work? Or does this need a new issue opened? (I have tried restarting the apps and mobile device.)
> 
> Thanks.

This sounds like a new issue. Please open a new issue for it and we will have a look.

**Comment by MRDGH2821** [comment]
I have 2 PCs & 1 mobile.
Sync between PC-1 & mobile works, sync between PC-2 & mobile works.

But when I try to add PC-1's QR code text into PC-2, PC-2 says that it is invalid. Vice-versa also has same result.

Is it intended to sync between PC<->Mobile only at the current moment?

**Comment by AppTester7** [comment]
It would be great if we could have the ability to (optionally) be able to Sync Subscription Notifications
When you have a YouTube channel one is subscribed to, that you have set to have 'notifications' set to ON, to be able to sync that notification status to other devices (desktop vs Android) either locally (Wi-Fi) or via a sort of cloud

**Comment by MRDGH2821** [comment]
> I have 2 PCs & 1 mobile. Sync between PC-1 & mobile works, sync between PC-2 & mobile works.
> 
> But when I try to add PC-1's QR code text into PC-2, PC-2 says that it is invalid. Vice-versa also has same result.
> 
> Is it intended to sync between PC<->Mobile only at the current moment?

In grayjay flatpak, I'm able to sync between PC<->PC!
